airflow/notifiers/__init__.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import jinja2
+import os
+
+from abc import ABCMeta, abstractmethod
+
+
+TEMPLATE_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'templates')
+
+
+class AirflowNotifier(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def send_notification(self, task_instance, state, message, **kwargs):
+        pass
+
+    def render_template(self, template, task_instance, **kwargs):
+        dag = task_instance.task.dag
+
+        jinja_loader = jinja2.FileSystemLoader([dag.folder, TEMPLATE_DIR])
```

Comment:
The dag.folder will actually help there. The `FileSystemLoader` will first try to load the template from there, and if it doesn't find it, fall back to the `TEMPLATE_DIR`. Changing a template there will indeed require a rebuild.
But, i'll add the option to pass a string as well, as that would indeed be usefull here.